ONSdigital/census-rm-pubsub
# Readiness indicator by file

## Motivation and Context
Without some indicator of readiness, Kubernetes considers the app ready as soon as its container is running, even if it is not properly configured and crashes soon after. The main downside to this is that deployments from concourse may go green when the container starts, even if the app is misconfigured and crashes immediately afterwards. This PR adds a file indicator which is only created once the app has completed its setup and is entering its message polling.

## What has changed
- Add a readiness context manager which creates and removes an indicator file
- Use readiness context manager on the main message polling loop after successful start up
- Add new environment variable for readiness file path

## How to test?
- Run the app properly configured and it should start up and create the readiness file. Then when it is stopped the file should be removed.

- Run the app with broken config (e.g. an incorrect topic name), it should fail to start and never create the readiness file at any point.

## Links
https://trello.com/c/UDZtgxSZ/483-deploy-census-rm-pubsub-to-kubernetes-8
